test/pylib/object_storage.py

```
@@ -94,14 +94,22 @@ def create_test_bucket(self, test_name: str):
+        A no-op once the bucket is gone, so a caller that cannot tell whether
+        its deferred destroy was registered may call this and let the callback
+        run too.
+        """
+        if not self.bucket_name:
+            return
         try:
             resource = self.get_resource()
             bucket = resource.Bucket(self.bucket_name)
             bucket.objects.all().delete()
             bucket.delete()
         except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)